File: front/lib/data/network/response_dto.dart
// http 통신의 결과로 받은 ResponseDTO를 검증하는 클래스
// ResponseDTO의 resultCode가 '1'인 경우(성공) data 객체를 반환
class ResponseDTO {
  static dynamic validation(Map<String, dynamic> response) {
    String resultCode = response['resultCode'];

    if (resultCode == '1') {
      // 성공
      return response['data'];
    } else if (resultCode == '2') {
      // 통신은 성공했지만 결과값이 성공이 아닌 경우
      return throw Exception('Server Data Error');
    }
  }
}

File: back/src/main/java/com/pingo/dto/ResponseDTO.java
package com.pingo.dto;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

// 공통 Response DTO

@Getter
@RequiredArgsConstructor(staticName = "of")
public class ResponseDTO<T> {
    private final String resultCode; // 1 성공, 2 실패
    private final String message;
    private final T data; // 반환값이 없는 경우에는 TRUE/FALSE

    @Override
    public String toString() {
        return "ResponseDTO{" +
                "resultCode='" + resultCode + '\'' +
                ", message='" + message + '\'' +
                ", data=" + data +
                '}';
    }
}
